applicationset/generators/git.go

```diff
@@ -238,6 +258,40 @@ func (g *GitGenerator) generateParamsFromGitFile(filePath string, fileContent []
 	return res, nil
 }
 
+// filterAppsFiles filters a list of file paths based on inclusion and exclusion patterns defined in GitFileGeneratorItems.
+// It uses doublestar globbing to support complex patterns like "**/*.yaml".
+func (g *GitGenerator) filterAppsFiles(files []argoprojiov1alpha1.GitFileGeneratorItem, allFilePaths []string) []string {
+	res := []string{}
+	for _, filePath := range allFilePaths {
+		appInclude := false
+		appExclude := false
+		for _, file := range files {
+			// since path.Match() doesn't natively support the ** double-star pattern for globbing
+			// we use the doublestar package - https://github.com/bmatcuk/doublestar
+			match, err := doublestar.Match(file.Path, filePath)
+			if err != nil {
+				log.WithError(err).WithField("itemPath", file).
+					WithField("path", filePath).Error("error while matching path to item path")
+				continue
```

Comment:
why continue over returning the error? i _think_ this would allow applications to be deleted if the match pattern was valid and then changed to be invalid.

Reply:
The error that you'd typically get will be the one when the file and patter match are unable to compile together. If we get an error, we simply move to check for the next pattern defined in the git generator path field. We simply shouldn't return the error because what if the file matches the next pattern.

Reply:
Yup! I understand what the code is doing and how the error would surface. What I am worried about is an operator/developer creating their applicationset with a valid pattern, having applications created. Then, they change their pattern to something invalid. Instead of an error surfacing, we are going to swallow the error, which may result in the deletion of applications.

Reply:
I lean yes, but wanted to get your opinion as well!

Reply:
can you please take a look now?